interactive_demo.py

```diff
@@ -0,0 +1,224 @@
+
+from rclpy.qos import QoSProfile, DurabilityPolicy
+from std_msgs.msg import String as StringMsg
+from tf2_ros import Buffer
+from tf2_ros import TransformListener
+from tf2_ros import TransformBroadcaster
+
+from pydrake.common import RandomDistribution
+from pydrake.systems.primitives import RandomSource
+
+
+class ForwardKinematics(LeafSystem):
+    """
+    Get forward kinematics of each joint on iiwa arm from manipulation station in world frame.
+
+    Offers something like the MultibodyPlant body_poses port.
+    """
+
+    def __init__(self, plant):
+        super().__init__()
+
+        # A multibody plant with a robot added to it
+        self._plant = plant
+        # Drake needs context to go along with a class instance
+        self._plant_context = self._plant.CreateDefaultContext()
+
+        num_positions = self._plant.num_positions()
+
+        # Input: List of joint positions matching order known by plant
+        self._joint_positions_port = self.DeclareVectorInputPort(
+            'joint_positions', BasicVector_[float](num_positions))
+        # Output: Rigid transforms of each body in base frame
+        self.DeclareAbstractOutputPort(
+            'transforms',
+            lambda: AbstractValue.Make([RigidTransform()]),
+            self._do_forward_kinematics)
+
+    def _do_forward_kinematics(self, context, data):
+        joint_positions = self._joint_positions_port.Eval(context)
+
+        # Set the latest positions in the plant context
+        self._plant.SetPositions(self._plant_context, joint_positions)
+
+        world_frame = self._plant.world_frame()
+
+        transforms = []
+        for i in range(self._plant.num_bodies()):
+            body = self._plant.get_body(BodyIndex(i))
+
+            # calculate pose of body in world frame
+            body_frame = body.body_frame()
+            transforms.append(
+                self._plant.CalcRelativeTransform(
+                    self._plant_context, world_frame, body_frame))
+
+        data.set_value(transforms)
+
+
+class PadVector(LeafSystem):
+    """
+    Add some constant values to pad a vector.
+    """
+
+    def __init__(self, num_inputs, num_outputs, output_value=0.0):
+        super().__init__()
+
+        if num_inputs > num_outputs:
+            raise ValueError('Must have more outputs than inputs')
+
+        self._difference = num_outputs - num_inputs
+        self._output_value = output_value
+
+        self._input_port = self.DeclareVectorInputPort(
+            'input', BasicVector_[float](num_inputs))
+
+        self.DeclareVectorOutputPort(
+            'output', BasicVector_[float](num_outputs), self._pad_output)
+
+    def _pad_output(self, context, output):
+        in_vec = self._input_port.Eval(context)
+        vector = [v for v in in_vec]
+        for i in range(self._difference):
+            vector.append(self._output_value)
+        output.SetFromVector(vector)
+
+
+if __name__ == '__main__':
+    builder = DiagramBuilder()
```

Comment:
This code is non-trivial, and should be in `main()` to avoid leaking out local variables.

Can you fix that here?

Reply:
wrapped in `main()` in 0fb2f29